### PR TITLE
Fix namelist for mpas_init

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

During testing with the mpas_app, I noticed a failure associated with an error when reading the namelist. This should fix that up.

**Type**

- [x] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [ ] Documentation
- [ ] Enhancement (adds a new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [ ] This is a breaking change (changes existing functionality)
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/drivers/mpas_init.py [hunk 1]
```diff
@@ -106,7 +107,7 @@ def namelist_file(self):
         update_values.setdefault("nhyd_model", {}).update(
             {
                 "config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
-                "config_run_duration": str_duration,
+                "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00"),
```
Comment (WeirAE) [inline src/uwtools/drivers/mpas_init.py:110]: This definitely looks better - do we need to update the tests to check this or similar though?

Comment (christinaholtNOAA) [reply src/uwtools/drivers/mpas_init.py:110]: I could add a test that ensures this is done.

File: src/uwtools/drivers/mpas_init.py [hunk 2]
```diff
@@ -94,8 +94,9 @@ def namelist_file(self):
         path = self._rundir / fn
         yield asset(path, path.is_file)
         yield None
-        duration = timedelta(hours=self._driver_config["boundary_conditions"]["length"])
-        str_duration = str(duration).replace(" days, ", "")
+        stop_time = self._cycle + timedelta(
+            hours=self._driver_config["boundary_conditions"]["length"]
+        )
         try:
```
Comment (NaureenBharwaniNOAA) [inline src/uwtools/drivers/mpas_init.py:100]: This also looks to be similar to what is in the `mpas` driver. Should we update the other as well?

Comment (christinaholtNOAA) [reply src/uwtools/drivers/mpas_init.py:100]: No, that was the problem. These two tools have distinctly different namelists. We went one step two far in making them similar.

Comment (NaureenBharwaniNOAA) [reply src/uwtools/drivers/mpas_init.py:100]: That makes sense!

**Review by WeirAE — APPROVED** [review]
LGTM, good fix!

**Review by nbharwani-cu — APPROVED** [review]
LGTM!

**Pull request merged**